### PR TITLE
refactor: Create infer subpackage to hold our statistics code

### PR DESCRIPTION
**Pull request by kratsg**
# Description

This will also be a `bumpversion/minor` update.

# Checklist Before Requesting Reviewer

- [x] Tests are passing
- [x] "WIP" removed from the title of the pull request
- [x] Selected an Assignee for the PR to be responsible for the log summary

# Before Merging

For the PR Assignees:

- [x] Summarize commit messages into a comprehensive review of the PR
```
* Create a pyhf.infer subpackage
* Move cli.stats to cli.infer
* Bump version: 0.2.2 → 0.3.0
```

### CODE REVIEW

**Comment by lukasheinrich** [comment]
this breaks the API, so probably should come with a versions bump

**Comment by matthewfeickert** [comment]
> hm didn't we first suggest `pyhf.infer` instead of `pyhf.stats`?

Ah, yes, we did. I forgot about that, but I agree that "infer" is better &mdash; it gets directly to what is being done, while "stats" is too broad.

**Comment by matthewfeickert** [comment]
> this breaks the API, so probably should come with a versions bump

Ah, so just doing it all in one fell swoop? Makes sense.

**Comment by kratsg** [comment]
Switched to `pyhf.infer`.

File: docs/governance/ROADMAP.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ The roadmap will be executed over mostly Quarter 3 of 2019 through Quarter 1 of
    - [x] Add tests using HEPData published sbottom likelihoods (Issue #518) [2019-Q3]
    - [x] Add CI with GitHub Actions and Azure Pipelines (PR #527, Issue #517) [2019-Q3]
    - [ ] Investigate rewrite of pytest fixtures to use modern pytest (Issue #370) [2019-Q3 → 2019-Q4]
-   - [ ] Factorize out the statistical fitting portion into `pyhf/stats/__init__.py` (PR #531) [2019-Q3 → 2019-Q4]
+   - [x] Factorize out the statistical fitting portion into `pyhf.infer/__init__.py` (PR #531) [2019-Q3 → 2019-Q4]
```
Comment (matthewfeickert) [inline docs/governance/ROADMAP.md]:
```suggestion
   - [x] Factorize out the statistical fitting portion into `pyhf.infer` (PR #531) [2019-Q3 → 2019-Q4]
```

File: docs/conf.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
 def setup(app):
     app.add_stylesheet(
-        'https://cdnjs.cloudflare.com/ajax/libs/github-fork-ribbon-css/0.2.2/gh-fork-ribbon.min.css'
```
Comment (matthewfeickert) [inline docs/conf.py]: I don't think this was supposed to get changed by bumpversion

Comment (kratsg) [reply docs/conf.py]: `[bumpversion:file:docs/conf.py]` technically, it is...

File: docs/conf.py [hunk 3]
```diff
@@ -24,7 +24,7 @@
...
+        'https://cdnjs.cloudflare.com/ajax/libs/github-fork-ribbon-css/0.3.0/gh-fork-ribbon.min.css'
```
Comment (matthewfeickert) [inline docs/conf.py]:
```suggestion
        'https://cdnjs.cloudflare.com/ajax/libs/github-fork-ribbon-css/0.2.2/gh-fork-ribbon.min.css'
```

**Pull request merged**